src/Platform/Microsoft.Testing.Extensions.HangDump/HangDumpActivityIndicator.cs

```diff
@@ -35,7 +35,7 @@ internal sealed class HangDumpActivityIndicator : IDataConsumer, ITestSessionLif
     private readonly ManualResetEventSlim _signalActivity = new(false);
     private readonly ManualResetEventSlim _mutexCreated = new(false);
     private readonly bool _traceLevelEnabled;
-    private readonly ConcurrentDictionary<string, (Type, DateTimeOffset)> _testsCurrentExecutionState = new();
+    private readonly ConcurrentDictionary<string, (string Name, Type Type, DateTimeOffset StartTime)> _testsCurrentExecutionState = new();
```

Comment:
Can we change the key type from string to UID for safety?

Reply:
yes.